scio/src/main/scala/org/apache/zeppelin/scio/DisplayHelpers.scala

```diff
@@ -0,0 +1,160 @@
+    "\n%html <font color=red>Result SCollection is empty!</font>\n"
+  private val maxResults = Integer.getInteger("zeppelin.scio.maxResult", 1000)
+  private[scio] val tab = "\t"
+  private[scio] val newline = "\n"
+  private[scio] val table = "%table"
+  private[scio] val rowLimitReachedMsg =
+    s"$newline<font color=red>Results are limited to " + maxResults + s" rows.</font>$newline"
+  private[scio] val bQSchemaIncomplete =
+    s"$newline<font color=red>Provided BigQuery Schema has not fields!</font>$newline"
+
+  private def notifyIfTruncated(it: Iterator[_]): Unit = {
+    if(it.hasNext) println(rowLimitReachedMsg)
+  }
+
+  /**
+   * Displays [[AnyVal]] values from given [[Iterator]].
+   */
+  private[scio] def displayAnyVal[T: ClassTag](it: Iterator[T], printer: (T) => String): Unit = {
+    if (it.isEmpty) {
+      println(sCollectionEmptyMsg)
+    } else {
+      println(s"$table value$newline${it.take(maxResults).map(printer).mkString(newline)}")
+      notifyIfTruncated(it)
+    }
+  }
+
+  /**
+   * Displays [[String]] values from given [[Iterator]].
+   */
+  private[scio] def displayString[T: ClassTag](it: Iterator[T], printer: (T) => String): Unit = {
+    if (it.isEmpty) {
+      println(sCollectionEmptyMsg)
+    } else {
+      println(s"$table value$newline${it.take(maxResults).map(printer).mkString(newline)}")
+      notifyIfTruncated(it)
+    }
+  }
+
+  /**
+   * Displays [[com.google.cloud.dataflow.sdk.values.KV]] values from given [[Iterator]].
+   */
+  private[scio] def displayKV[K: ClassTag, V: ClassTag](it: Iterator[(K,V)]): Unit = {
+    if (it.isEmpty) {
+      println(sCollectionEmptyMsg)
+    } else {
+      val content = it.take(maxResults).map{ case (k, v) => s"$k$tab$v" }.mkString(newline)
+      println(s"$table key${tab}value$newline$content")
+      notifyIfTruncated(it)
+    }
+  }
+
+  /**
+   * Displays [[Product]] values from given [[Iterator]].
+   */
+  private[scio] def displayProduct[T: ClassTag](it: Iterator[T])
+                                               (implicit ev: T <:< Product): Unit = {
+    if (it.isEmpty) {
+      println(sCollectionEmptyMsg)
+    } else {
+      val first = it.next()
+      //TODO is this safe field name to value iterator?
+      val fieldNames = first.getClass.getDeclaredFields.map(_.getName)
+
+      val header = fieldNames.mkString(tab)
+      val firstStr = first.productIterator.mkString(tab)
+      val content = it.take(maxResults - 1).map(_.productIterator.mkString(tab)).mkString(newline)
+      println(s"$table $header$newline$firstStr$newline$content")
+      notifyIfTruncated(it)
+    }
+  }
+
+  /**
+   * Displays Avro values from given [[Iterator]] using optional [[Schema]].
+   * @param schema optional "view" schema, otherwise schema is inferred from the first object
+   */
+  private[scio] def displayAvro[T: ClassTag](it: Iterator[T], schema: Schema = null)
+                                            (implicit ev: T <:< GenericRecord): Unit = {
```

Comment:
Starting from 1.7.5 SpecificRecordBase implements GenericRecord (~2013). In here we use 1.7.7. That said, it might worth supporting old versions and use IndexedRecord.